src/js/navIcons.js

```diff
@@ -30,17 +30,20 @@ https://github.com/fluid-project/infusion/raw/master/Infusion-LICENSE.txt
             show: "gpii-fd-show"
         },
         modelListeners: {
-            "isActive.setState": {
-                listener: "gpii.firstDiscovery.icon.setState",
-                args: ["{that}", "{change}.value", "{change}.oldValue"]
+            "isActive.setActiveState": {
+                "this": "{that}.container",
+                method: "toggleClass",
+                args: ["{that}.options.styles.active", "{change}.value"]
+            },
+            "isConfirmed.setVisitedState": {
+                listener: "gpii.firstDiscovery.icon.setVisitedState",
+                args: ["{that}", "{change}.value"]
             }
         }
     });
 
-    gpii.firstDiscovery.icon.setState = function (that, isActive, isActivePrev) {
-        that.container.toggleClass(that.options.styles.active, isActive);
-
-        if (isActivePrev && !isActive) {
+    gpii.firstDiscovery.icon.setVisitedState = function (that, isConfirmed) {
```

Comment:
`setVisitedState` should probably be renamed to `setConfirmedState` and same for all other similar occurrences.